bindings/java/src/org/sleuthkit/datamodel/AbstractFile.java

```diff
@@ -90,6 +94,8 @@
 	private static final ResourceBundle BUNDLE = ResourceBundle.getBundle("org.sleuthkit.datamodel.Bundle");
 	private long dataSourceObjectId;
 	private final String extension;
+	private final List<Attribute> fileAttributesCache = new ArrayList<Attribute>();
+	private final AtomicBoolean loadedAttributesCacheFromDb = new AtomicBoolean(false);
```

Comment:
The flag for the cache is thread safe, but the cache itself is not. Which policy do we need?

Discussion to follow via email.